raiden-ts/src/channels/epics.ts

```diff
@@ -62,52 +62,81 @@ export const initNewBlockEpic = (
   );
 
 /**
- * Monitor registry for new token networks and monitor them
+ * On first run, scan registry and token networks for registered TokenNetworks of interest
+ * (ones which has/had channels with us) and monitors them. On next runs, just monitors the
+ * previously monitored ones.
  *
  * @param action$ - Observable of RaidenActions
  * @param state$ - Observable of RaidenStates
  * @param registryContract,contractsInfo - RaidenEpicDeps members
  * @returns Observable of tokenMonitored actions
  */
-export const initMonitorRegistryEpic = (
+export const initTokensRegistryEpic = (
   {}: Observable<RaidenAction>,
   state$: Observable<RaidenState>,
-  { registryContract, contractsInfo }: RaidenEpicDeps,
+  { address, provider, registryContract, contractsInfo }: RaidenEpicDeps,
 ): Observable<tokenMonitored> =>
   state$.pipe(
-    publishReplay(1, undefined, state$ =>
-      state$.pipe(
-        first(),
-        switchMap(state =>
-          merge(
-            // monitor old (in case of empty tokens) and new registered tokens
-            // and starts monitoring every registered token
-            getEventsStream<[Address, Address, Event]>(
-              registryContract,
-              [registryContract.filters.TokenNetworkCreated(null, null)],
-              isEmpty(state.tokens)
-                ? of(contractsInfo.TokenNetworkRegistry.block_number)
-                : undefined,
-            ).pipe(
-              withLatestFrom(state$),
-              map(([[token, tokenNetwork, event], state]) =>
-                tokenMonitored({
-                  token,
-                  tokenNetwork,
-                  fromBlock: !(token in state.tokens) ? event.blockNumber : undefined,
-                }),
-              ),
-            ),
-            // monitor previously monitored tokens
-            from(Object.entries(state.tokens)).pipe(
-              map(([token, tokenNetwork]) =>
-                tokenMonitored({ token: token as Address, tokenNetwork }),
+    take(1),
+    mergeMap(state => {
+      const encodedAddress = defaultAbiCoder.encode(['address'], [address]);
+      // if tokens are already initialized, use it
+      if (!isEmpty(state.tokens))
```

Comment:
What happens when token addresses in the token registry change (like in #833)?

Reply:
@nephix Token addresses in the registry don't change. If you have two tokens with the same symbol and different address they are different tokens. The only unique identifier of a token is the address. In the case of TTT appearing with different addresses between different states, my guess would be that this is a different deployment of the smart contracts which would mean that this is a different token registry, which means that this is a different "Network".

Reply:
Hm ok that makes sense, thanks for clarifying 🙏